src/TeamSpeak3QueryApi/Query/Protocols/SshProtocol.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Renci.SshNet;
+using Renci.SshNet.Common;
+using TeamSpeak3QueryApi.Net.Enums;
+using TeamSpeak3QueryApi.Net.Extensions;
+
+namespace TeamSpeak3QueryApi.Net.Query.Protocols
+{
+    public class SshProtocol : QueryClient, IProtocol
```

Comment:
I'd name this `SshQueryClient` and the other one `TelnetQueryClient` or `RawQueryClient` (which is technically more correct).

Reply:
Confirmed. It´s much better of it´s extends from a query client.